### ISSUE TITLE
Mach Ports?

### ISSUE THREAD

**Issue by timdream** [matryer/bitbar#458]
Hi, I am trying to figuring out the cause of the memory leak. I can reduce the memory leak by changing the interval of the script invocation. One number that keeps growing is on the "Ports" column.

<img width="1052" alt="screen shot 2018-01-29 at 11 39 56 pm" src="https://user-images.githubusercontent.com/93093/35519918-922f847a-0550-11e8-8511-55b5c55ed504.png">

How can I debug this further? Thanks.

**Comment by gingerbeardman** [comment]
how long does your script take to execute?

how frequently is it being updated?

**Comment by timdream** [comment]
```
caffeinate.1h.sh
date.1m.js
hid-active.10s.sh
ssid-display-sleep.30s.sh
```

Would it be possible the problem is caused by the script themselves? Should I pastebin them here?

**Comment by timdream** [comment]
They are all fairly quick:

```
$ time ./date.1m.js > /dev/null

real	0m0.107s
user	0m0.074s
sys	0m0.027s
$ time ./caffeinate.1h.sh > /dev/null

real	0m0.026s
user	0m0.010s
sys	0m0.012s
$ time ./hid-active.10s.sh > /dev/null

real	0m0.069s
user	0m0.028s
sys	0m0.037s
$ time ./ssid-display-sleep.30s.sh > /dev/null

real	0m0.074s
user	0m0.042s
sys	0m0.042s
```

**Comment by gingerbeardman** [comment]
is the leak coming from one script in particular or from BitBar itself?

try removing one script at a time to see if the leak persists.

**Comment by oleander** [comment]
When I ported the application to swift I found a few memory leaks in the current application. The only one I can remember now is one related to downloading images. 

It’s difficult to tell wherever the problem is on your end or bitbar, but it’s not unlikely that the problem is caused directly by bitbar. 

Not sure if that helped, but now you know :)

**Comment by timdream** [comment]
After trying to isolating the cause, I can see the "Ports" count going up even with only a small `test.5s.sh` script like

```shell
#!/bin/bash

echo test
```

Memory also slightly increased from 19M to around 25M. What should I do to look further? Thanks :)

**Comment by gingerbeardman** [comment]
My testing:

- @timdream test script
- filename test.5s.sh
- only one script

i do not see any leaks.

CPU 0.1%-0.4%
threads 6
ports hovering ~180-190
memory static at 7.5MB

Mac OS X 10.13.3 (17D47)

**Comment by timdream** [comment]
The only difference here is a different plug-in dir and the macOS version (10.12.6 (16G1212)).

**Comment by gingerbeardman** [comment]
> different plug-in dir 

what do you mean?

mine is located at ~/Library/BitBar/ and contains only the one test script.

**Comment by timdream** [comment]
Yeah I put my scripts in `~/.rc/Darwin/BitBar/` so I can manage them in a git repo.

**Comment by gingerbeardman** [comment]
Hmm. Try another directory without the . character?

And I'll try your path

**Comment by timdream** [comment]
Didn't work. There are other quirks as well

* Check for Updates always return an error.
* Deleting `~/Library/Preferences/com.matryer.BitBar.plist` doesn't reset the plug-in dir.

How do I properly reset all settings?

**Comment by gingerbeardman** [comment]
I also get an error checking for updates.

I've used `defaults delete com.matryer.BitBar` to reset plugins directory.

**Comment by gingerbeardman** [comment]
Great!
